execution_requests.go

```diff
@@ -0,0 +1,48 @@
+package compute
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func (c *Client) Exec(runnable *Runnable, body io.Reader) ([]byte, *http.Response, error) {
+	req, err := c.execRequestBuilder(http.MethodPost, runnable.VersionPath(), body)
+
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to Client.Exec")
+	}
+
+	res, err := c.do(req)
+	if err != nil {
+		return nil, res, errors.Wrap(err, "failed to Client.Exec")
+	}
+
+	result, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, res, errors.Wrap(err, "failed to Client.Exec")
+	}
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
With the change to `c.do`, I don't think this will work as expected any more?

Reply:
Hmm yeah, I'll have to put the status code check first.